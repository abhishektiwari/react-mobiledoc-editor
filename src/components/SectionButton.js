import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';
import titleCase from '../utils/titleCase';

const SectionButton = ({ tag = '', type = 'button', children = titleCase(tag), className, activeClassName = 'active', ...props }, { editor, activeSectionTags = []}) => {
  const onClick = () => editor.toggleSection(tag);
  className = classNames(className, {
    [activeClassName]: activeSectionTags.indexOf(tag.toLowerCase()) > -1
  });
  props = { type, ...props, onClick, className };
  return <button { ...props }>{children}</button>;
};

SectionButton.propTypes = {
  tag: PropTypes.string.isRequired,
  children: PropTypes.node
};

SectionButton.contextTypes = {
  editor: PropTypes.object,
  activeSectionTags: PropTypes.array
};

export default SectionButton;
