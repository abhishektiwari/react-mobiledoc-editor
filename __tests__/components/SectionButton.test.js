import React from 'react';
import { shallow } from 'enzyme';
import SectionButton from '../../src/components/SectionButton';

describe('<SectionButton />', () => {
  it('should render', () => {
    const wrapper = shallow(<SectionButton tag="UL" type="reset" title="List" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render children', () => {
    const wrapper = shallow(<SectionButton tag="UL">List</SectionButton>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should toggle markup on click', () => {
    const editor = { toggleSection: jest.fn() };
    const context = { editor };
    const wrapper = shallow(<SectionButton tag='UL' />, { context });
    wrapper.find('button').simulate('click');
    expect(editor.toggleSection).toBeCalledWith('UL');
  });

  it('should set active class', () => {
    const context = { activeSectionTags: ['ul']};
    const wrapper = shallow(<SectionButton tag='UL' className="keep" />, { context });
    expect(wrapper).toMatchSnapshot();
  });
});
