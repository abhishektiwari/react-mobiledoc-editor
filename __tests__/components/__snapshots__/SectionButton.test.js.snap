// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionButton /> should render 1`] = `
<button
  className=""
  onClick={[Function]}
  title="List"
  type="reset"
>
  UL
</button>
`;

exports[`<SectionButton /> should render children 1`] = `
<button
  className=""
  onClick={[Function]}
  type="button"
>
  List
</button>
`;

exports[`<SectionButton /> should set active class 1`] = `
<button
  className="keep active"
  onClick={[Function]}
  type="button"
>
  UL
</button>
`;
